### PR TITLE
HOTFIX - Updated Permission Logic

### PR DESCRIPTION
**Pull request by k-macmillan**
# Description

This started as an [UnboundLocalError](https://docs.python.org/3/library/exceptions.html#UnboundLocalError) due to a variable not being set. There were some issues with the logic that needed fixing as well. Cleanup here and there, and some new test cases.

issue N/A

## How Has This Been Tested?

Wrote new unit tests and deployed to dev, perf, and staging with the feature flag on and off. All unit and regression tests pass.

## Checklist

- [x] I have assigned myself to this PR
- [x] PR has an [appropriate title](https://github.com/department-of-veterans-affairs/vanotify-team/blob/main/Engineering/team_agreement.md#naming-prs): #9999 - What the thing does
- [x] PR has a detailed description, including links to specific documentation
- [x] I have added the [appropriate labels](https://github.com/department-of-veterans-affairs/vanotify-team/blob/main/Engineering/pull_request_labels.md#vanotify-labels) to the PR.
- [x] I did not remove any parts of the template, such as checkboxes even if they are not used
- [x] My code follows the [style guidelines](https://github.com/department-of-veterans-affairs/vanotify-team/blob/main/Process/style_guide.md) of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to any documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works. [Testing guidelines](https://github.com/department-of-veterans-affairs/vanotify-team/blob/main/Process/testing_guide.md)
- [x] I have ensured the latest main is merged into my branch and all checks are green prior to review
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] The ticket was moved into the DEV test column when I began testing this change

### CODE REVIEW

**Review by cris-oddball — APPROVED** [review]

File: app/va/va_profile/va_profile_client.py [hunk 1]
```diff
@@ -326,8 +335,14 @@ def get_is_communication_allowed_from_profile(
...
-                assert isinstance(perm['allowed'], bool)
-                return perm['allowed']
+                # if default send is true and allowed is false, return false
+                # if default send is true and allowed is true, return true
+                # if default send is false, default to what it finds
+                permission = perm['allowed']
+                if permission is not None:
+                    return perm['allowed']
+                else:
+                    return notification.default_send
```
Comment (kalbfled) [inline app/va/va_profile/va_profile_client.py:345]: Ditto.
```suggestion
                return perm.get('allowed', notification.default_send)
```

File: tests/app/celery/test_contact_information_tasks.py [hunk 2]
```diff
@@ -314,3 +319,68 @@ def test_exception_sets_failure_reason_if_thrown(
...
+        [True, False, False],
+        [False, True, True],
+        [False, False, False],
+        # If the user has not set a preference, go with the default_send
+        [True, None, True],
+        [False, None, False],
+    ],
+)
+@pytest.mark.parametrize('notification_type', [CommunicationChannel.EMAIL, CommunicationChannel.TEXT])
+def test_ut_get_email_or_sms_with_permission_utilizes_default_send(
```
Comment (kalbfled) [inline tests/app/celery/test_contact_information_tasks.py:338]: There's no problem with this, but we discussed that we're not distinguishing between "ut", etc., anymore.

File: tests/app/celery/test_contact_information_tasks.py [hunk 3]
```diff
@@ -314,3 +319,68 @@ def test_exception_sets_failure_reason_if_thrown(
...
+    communication_item = sample_communication_item(default_send)
+    template = sample_template(template_type=channel, communication_item_id=communication_item.id)
+
+    notification = sample_notification(
+        template=template,
+        gen_type=channel,
+        recipient_identifiers=[{'id_type': IdentifierType.VA_PROFILE_ID.value, 'id_value': '1234'}],
+    )
+
+    # if user_set is not None:
```
Comment (kalbfled) [inline tests/app/celery/test_contact_information_tasks.py:362]: Delete?

File: tests/app/celery/test_contact_information_tasks.py [hunk 4]
```diff
@@ -314,3 +319,68 @@ def test_exception_sets_failure_reason_if_thrown(
...
+    profile['communicationPermissions'][0]['allowed'] = user_set
+    profile['communicationPermissions'][0]['communicationItemId'] = notification.va_profile_item_id
+    profile['communicationPermissions'][0]['communicationChannelId'] = notification_type.id
+
+    # contact_info = sample_contact_information(allowed=False)
```
Comment (kalbfled) [inline tests/app/celery/test_contact_information_tasks.py:367]: Delete?

**Review by kalbfled — CHANGES_REQUESTED** [review]

File: tests/app/celery/test_contact_information_tasks.py (hunk 2 above)
Comment (k-macmillan) [reply tests/app/celery/test_contact_information_tasks.py:338]: Sure, I can delete those. Was copy/paste.

**Comment by k-macmillan** [comment]
Changes were comments and/or unit test names. An "e" was added to a raise statement to alleviate any confusion, but does not change the code at all. This is ready for a re-review.

**Review by cris-oddball — APPROVED** [review]

**Review by kalbfled — APPROVED** [review]

**Pull request merged**